qa/rpc-tests/addressindex.py

```diff
@@ -0,0 +1,389 @@
+
+        mempool = self.nodes[0].getaddressmempool({'addresses': [b, a]})
+        assert_equal(len(mempool), 0)
+
+        bal = self.nodes[2].getaddressbalance(a)
+        assert_equal(bal['received'], expected*COIN)
+        # the value 4 UTXO is no longer in our balance
+        assert_equal(bal['balance'], (expected-4)*COIN)
+
+        # Ensure the change from that transaction appears
+        tx = self.nodes[0].getrawtransaction(txid, 1)
+        change_vout = filter(lambda v: v['valueZat'] != 3*COIN, tx['vout'])
+        change = change_vout[0]['scriptPubKey']['addresses'][0]
+        bal = self.nodes[2].getaddressbalance(change)
+        assert(bal['received'] > 0)
+        assert(bal['received'] < (4-3)*COIN)
+        assert_equal(bal['received'], bal['balance'])
+        assert_equal(self.nodes[2].getaddresstxids(change), [txid])
+
+        # Further checks that limiting by height works
+
+        # non-overlapping range returns an empty list
+        height_txids = self.nodes[1].getaddresstxids({
+            'addresses': [a],
+            'start': 106,
+            'end': 105
+        })
+        assert_equal(height_txids, [])
+
+        # various ranges
+        for i in range(5):
+            height_txids = self.nodes[1].getaddresstxids({
+                'addresses': [a],
+                'start': 106,
+                'end': 106+i
+            })
+            assert_equal(height_txids, txids_a[0:i+1])
+
+        height_txids = self.nodes[1].getaddresstxids({
+            'addresses': [a],
+            'start': 0,
+            'end': 108
+        })
+        assert_equal(height_txids, txids_a[0:3])
+        # end=0 means return the entire range
+        height_txids = self.nodes[1].getaddresstxids({
+            'addresses': [a],
+            'start': 107,
+            'end': 0
+        })
+        assert_equal(height_txids, txids_a)
+
+        # Further check specifying multiple addresses
+        txids_all = list(txids_a)
+        txids_all += self.nodes[1].getaddresstxids(addr_p2pkh)
+        txids_all += self.nodes[1].getaddresstxids(addr_p2sh)
+        multitxids = self.nodes[1].getaddresstxids({
+            'addresses': [a, addr_p2sh, addr_p2pkh]
+        })
+        # No dups in return list from getaddresstxids
+        assert_equal(len(multitxids), len(set(multitxids)))
+
+        # set(txids_all) removes its (expected) duplicates
+        assert_equal(set(multitxids), set(txids_all))
+
+        deltas = self.nodes[1].getaddressdeltas({
+            'addresses': [a],
+        })
+        assert_equal(len(deltas), len(expected_deltas))
+        for z in zip(deltas, expected_deltas):
+            assert_equal(z[0]['address'],   a)
+            assert_equal(z[0]['height'],    z[1]['height'])
+            assert_equal(z[0]['satoshis'],  z[1]['satoshis'])
+            assert_equal(z[0]['txid'],      z[1]['txid'])
+
+        deltas_limited = self.nodes[1].getaddressdeltas({
```

Comment:
If we wanted to be able to do this in a single line, we could add the following function:
```
def getaddressdeltas(node, addresses, start, end, chainInfo=None):
    params = {
        'addresses': addresses,
        'start': start,
        'end': end,
    }
    if chainInfo is not None:
        params.update({'chainInfo': chainInfo})
    return node.getaddressdeltas(params)
```